htp/htp_transaction.c

```diff
@@ -789,6 +789,25 @@ static htp_status_t htp_tx_res_process_body_data_decompressor_callback(htp_tx_da
     // Invoke all callbacks.
     htp_status_t rc = htp_res_run_hook_body_data(d->tx->connp, d);
     if (rc != HTP_OK) return HTP_ERROR;
+    d->tx->connp->out_decompressor->nb_callbacks++;
+    if (d->tx->connp->out_decompressor->nb_callbacks % HTP_COMPRESSION_TIME_FREQ_TEST == 0) {
+        struct timeval after, duration;
+        gettimeofday(&after, NULL);
+        // sanity check for race condition if system time changed
+        if ( timercmp(&after, &d->tx->connp->out_decompressor->time_before, >) ) {
+            timersub(&after, &d->tx->connp->out_decompressor->time_before, &duration);
```

Comment:
Looks like I've put you on the wrong path here. `timeradd` and `timersub` are not available on all of our platforms, It's missing on Windows (MinGW). `timercmp` *is* available already in use in Suricata in a few places.

We could see if we can add in a properly licensed implementation from somewhere, or we can do back to code that doesn't use these calls.

Reply:
I will go back to first version where we do not use these functions